raiden/transfer/state.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+from raiden.transfer.architecture import State
+from raiden.utils import pex
+# pylint: disable=too-few-public-methods,too-many-arguments,too-many-instance-attributes
+
+
+class RouteState(State):
+    """ Route state.
+
+    Args:
+        state (string): The current state of the route (available or
+            unavailable).
+        node_address (address): The address of the next_hop.
+        channel_address (address): The address of the on chain netting channel.
+        available_balance (int): The current available balance that can be transferred
+            through `node_address`.
+        settle_timeout (int): The settle_timeout of the channel set in the
+            smart contract.
+        reveal_timeout (int): The channel configured reveal_timeout.
+        close_block (Nullable[int]): None if the channel is open, otherwise
+            the block number at which the channel was closed.
+    """
+    __slots__ = (
+        'state',
+        'node_address',
+        'channel_address',
+        'available_balance',
+        'settle_timeout',
+        'reveal_timeout',
+        'close_block',
+    )
+
+    valid_states = (
```

Comment:
Sadly python doesn't have static type checking so an enum wouldn't fix the problem by itself.

We could always use these values as "globals" instead of the typing the strings, this sort of coding guidelines would make checking easier. I didn't do it out of laziness.